distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Day_Of_Week.enso

```diff
@@ -0,0 +1,52 @@
+from Standard.Base import all
+
+type Day_Of_Week
+    type Sunday
+
+    type Monday
+
+    type Tuesday
+
+    type Wednesday
+
+    type Thursday
+
+    type Friday
+
+    type Saturday
+
+    ## Convert the Day_Of_Week to an Integer
+
+       Arguments:
+       - `first_day`: The first day of the week.
+       - `include_zero`: If True, first day of the week is 0 otherwise is 1.
+    to_integer : Day_Of_Week -> Boolean -> Integer
+    to_integer self first_day=Sunday start_at_zero=False =
+        day_number = case self of
+            Sunday -> 0
+            Monday -> 1
+            Tuesday -> 2
+            Wednesday -> 3
+            Thursday -> 4
+            Friday -> 5
+            Saturday -> 6
+
+        shifted = if first_day == Day_Of_Week.Sunday then day_number else
+            (day_number + 7 - (first_day.to_integer start_at_zero=True)) % 7
```

Comment:
I hope that it is tested for different combinations of `first_day` and `start_at_zero`